distributed/semaphore.py

```diff
@@ -37,41 +39,45 @@ class SemaphoreExtension:
     * semaphore_acquire
     * semaphore_release
     * semaphore_create
+    * semaphore_close
+    * semaphore_refresh_leases
     """
 
     def __init__(self, scheduler):
         self.scheduler = scheduler
-        self.leases = defaultdict(deque)
         self.events = defaultdict(asyncio.Event)
         self.max_leases = dict()
-        self.leases_per_client = defaultdict(partial(defaultdict, deque))
+        self.leases = defaultdict(dict)
```

Comment:
```suggestion
        # `self.leases` is used to track each lease of each semaphore "client" with their latest "refresh" timestamp
        # Structure: {"client_name_1" : {"lease_id_1": `lease_timestamp_1`, "lease_id_2": ...},  "client_name_2" : {...}}
        self.leases = defaultdict(dict)  
```

Reply:
FYI: The structure you're suggesting here is wrong. It is `semaphore_name.lease_id.timestamp`